presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoMetadata.java

```diff
@@ -95,11 +95,17 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     }
 
     @Override
-    public List<SchemaTableName> listTables(ConnectorSession session, String schemaNameOrNull)
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaNameOrNull)
     {
-        ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
+        List<String> schemaNames = schemaNameOrNull.map(ImmutableList::of)
+                .orElseGet(() -> (ImmutableList<String>) listSchemaNames(session));
+        return listTables(schemaNames);
```

Comment:
It appears that we can easily merge these two methods (top part computes `schemaNames`, bottom part iterates over it)